### PR TITLE
Wrap correction functions for mc optional config

### PR DESCRIPTION
**Pull request by zhut19**
Before you submit this PR: make sure to put all operations-related information in a wiki-note, a PR should be about code and is publicly accessible

**What is the problem / what does the code in this PR do**
Following the discussion in https://github.com/XENONnT/WFSim/pull/108 of using the alternative id when using CMT, we would like to make it possible through this PR.

**Can you briefly describe how it works?**
The  `correction_options` function wrapper would look for 'MC' tag in the second argument  (which generally is the `*model` context config) and use the alternative run_id included in the config to replace the original run_id in the get correction functions.

**Can you give a minimal working example (or illustrate with a figure)?**
```
to_pe = straxen.get_to_pe(run_id, ('MC', cmt_run_id, 'CMT_model', ('to_pe_model', 'ONLINE')), st.config['n_tpc_pmts'])
```

will return to_pe values for the run with cmt_run_id instead of the run with run_id.

### CODE REVIEW

**Review by ahiguera-mx — APPROVED** [review]
Hi @zhut19 
Thanks for putting all of this together, I've tested and everything seems to be working accordingly.

**Review by jorana — CHANGES_REQUESTED** [review]
Hi Tianyu, thanks a lot. Looks like an elegant solution.

I've written a small test of how I would (based on your description) use this for elife. It appears to not fully work due to the `*args` argument.

**General comment on cmt-code**
One concern is that drafting CMT configs now becomes an art of it's own. We now get tuples with 7 positional arguments and 3 level nesting of tuples which is bound to cause errors somewhere. 

Perhaps not an issue for discussion in this PR, but I think having CMT being registered and initialized as a context attribute would greatly simplify things. This will require some work, also on the strax side but you would do the init once and then do not have to worry about it too much anymore:

```python
st=strax.Context(
   corrections=straxen.CorrectionsManagementServices(detector= 'nT' | '1T' | 'mc_nT | ...)),
   storage = [...]
```

**Comment by ahiguera-mx** [comment]
Hi @jorana 
Thanks for the useful comments 
> Hi Tianyu, thanks a lot. Looks like an elegant solution.
> 
> I've written a small test of how I would (based on your description) use this for elife. It appears to not fully work due to the `*args` argument.

I've tested also elife and work for me here is the command line I used 
`straxen.get_correction_from_cmt('010126', ('MC','013120' ,'elife','ONLINE',True))`

> _edit_: you need to run the test locally for it to fail, since this PR is from a remote fork, you don't test with access to the database which is needed to access CMT.
I've done that and it works 
> 
> **General comment on cmt-calls**
> One concern is that drafting CMT configs now becomes an art of it's own. We now get tuples with 7 positional arguments and 2 level nesting of tuples which is bound to cause errors somewhere.

The "new configs" are only needed for WFsim, so no changes to the current configs in the plugins. Once this is implemented I can work on standardize  the configs
> 
> Perhaps not an issue for discussion in this PR, but I think having CMT being registered and initialized as a context attribute would greatly simplify things. This will require some work, also on the strax side but you would do the init once and then do not have to worry about it too much anymore:

This is also in my to do list as we have discussed 
> 
> ```python
> st=strax.Context(
>    corrections=straxen.CorrectionsManagementServices(detector= 'nT' | '1T' | 'mc_nT | ...)),
>    storage = [...]
> ```

File: straxen/get_corrections.py [hunk 1]
```diff
@@ -141,6 +168,7 @@ def get_config_from_cmt(run_id, conf):
     return this_file
 
 
+@correction_options
```
[the remark this answers is not in the document]
Comment (ahiguera-mx) [reply straxen/get_corrections.py:171]: Good catch

**Review by jorana — COMMENTED** [review]
@ahiguera-mx , thanks, great indeed now it works I just got the tuple brackets wrong (substantiating my comment above).

I agree, it's nice to add this now for MC and it's great that you are on top of these things! Thanks both!

File: straxen/get_corrections.py [hunk 2]
```diff
@@ -2,12 +2,37 @@
...
+    Search for special options like ["MC", ] and apply arg shuffling accordingly
+    :param get_correction_function: A function here in the get_corrections module
+    :returns: The function wrapped with the option search
+    """
+    @wraps(get_correction_function)
+    def correction_options_wrapped(run_id, conf, *arg):
+        if isinstance(conf, tuple):
+            # MC chain simulation can put run_id inside configuration
+            if 'MC' in conf:
+                tag, run_id, *conf = conf
```
[the remark this answers is not in the document]
Comment (zhut19) [reply straxen/get_corrections.py:22]: Maybe call this one cmt_run_id, since the other one is mc_run_id

[the remark this answers is not in the document]
Comment (jorana) [reply straxen/get_corrections.py:22]: 🤣 oh, you are right, was too quick

**Pull request merged**